src/main/scala/codacy/docker/api/Tool.scala

```diff
@@ -3,7 +3,8 @@ package codacy.docker.api
 import scala.util.Try
 
 trait Tool {
-  def apply(source: Source.Directory, configuration: Option[List[Pattern.Definition]], files: Option[Set[Source.File]])
```

Comment:
I think it should be better to create an override method that calls this one with empty options to avoid breaking changes